internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go

```diff
@@ -201,8 +202,22 @@ func (*Sequencer) Boot(r runtime.Runtime) []runtime.Phase {
 	phases := PhaseList{}
 
 	wipe := procfs.ProcCmdline().Get(constants.KernelParamWipe).First()
-	if wipe != nil && *wipe == "system" {
-		return phases.Append("wipeSystemDisk", ResetSystemDisk).Append("reboot", Reboot)
+	if wipe != nil {
+		switch *wipe {
+		case "system":
+			return phases.Append("wipeSystemDisk", ResetSystemDisk).Append("reboot", Reboot)
+		case "system:EPHEMERAL,STATE":
```

Comment:
same here - constants?

Reply:
in fact, we don't need to reboot here.. we could move this whole block to be significantly earlier, in the `Initialize` sequence right before `MountStatePartition` 

if we do a full wipe, we have to reboot, if we do selective wipe and we don't wipe BOOT, we can continue on - Talos will act on it.

e.g. we could write a task which calls existing tasks as needed?

Reply:
That can be arranged actually...